toolchain/check/import_ref.cpp

```diff
@@ -879,11 +881,26 @@ class ImportRefResolver {
          .definition_id = function.definition_id.is_valid()
                               ? function_decl_id
                               : SemIR::InstId::Invalid});
+    function_decl.type_id = context_.GetFunctionType(function_decl.function_id);
     // Write the function ID into the FunctionDecl.
     context_.ReplaceInstBeforeConstantUse(function_decl_id, function_decl);
     return {context_.constant_values().Get(function_decl_id)};
   }
 
+  auto TryResolveTypedInst(SemIR::FunctionType inst) -> ResolveResult {
+    auto initial_work = work_stack_.size();
+    CARBON_CHECK(inst.type_id == SemIR::TypeId::TypeType);
+    auto fn_const_id = GetLocalConstantId(
+        import_ir_.functions().Get(inst.function_id).decl_id);
+    if (HasNewWork(initial_work)) {
+      return ResolveResult::Retry();
+    }
+    auto fn_val =
+        context_.insts().GetAs<SemIR::StructValue>(fn_const_id.inst_id());
+    CARBON_CHECK(context_.types().Is<SemIR::FunctionType>(fn_val.type_id));
+    return {context_.types().GetConstantId(fn_val.type_id)};
```

Comment:
The `GetAs` seems redundant here, given we don't need any of the `StructValue`-specific fields:
```suggestion
    auto fn_val = context_.insts().Get(fn_const_id.inst_id());
    CARBON_CHECK(context_.types().Is<SemIR::FunctionType>(fn_val.type_id()));
    return {context_.types().GetConstantId(fn_val.type_id())};
```